base/applications/osk/main.c

```diff
@@ -585,12 +585,35 @@ int WINAPI wWinMain(HINSTANCE hInstance,
                     int show)
 {
     HANDLE hMutex;
+    DWORD dwError;
     INT LayoutResource;
 
     UNREFERENCED_PARAMETER(prev);
     UNREFERENCED_PARAMETER(cmdline);
     UNREFERENCED_PARAMETER(show);
 
+    /*
+        Obtain a mutex for the program. This will ensure that
+        the program is launched only once.
+    */
+    hMutex = CreateMutexW(NULL, FALSE, L"OSKRunning");
+
+    if (hMutex)
```

Comment:
I agree that failure to create the mutex should be ignored, but perhaps add a DPRINT or something?

Reply:
Of course. I'll do that when I amend the commit later.

Reply:
Not sure we really need to care about that; it's just a user-mode app, and not the kernel where such conditions need to not happen in general.